src/components/Initialization/Initialization.js

```diff
@@ -38,27 +39,81 @@ class Initialization extends PureComponent<Props, State> {
     logger.logEvent('load-application', {
       node_version: nodeVersion,
     });
-    window.addEventListener('beforeunload', this.killAllRunningProcesses);
-  }
 
-  componentWillUnmount() {
-    window.removeEventListener('beforeunload', this.killAllRunningProcesses);
+    ipcRenderer.on('app-will-close', this.appWillClose);
   }
 
-  killAllRunningProcesses = () => {
+  appWillClose = () => {
+    const { isQueueEmpty, queue, projects } = this.props;
+
+    const activeActions = Object.keys(queue).map(projectId => ({
+      name: projects.find(project => project.id === projectId).name,
+      pending: queue[projectId],
+    }));
+
+    if (!isQueueEmpty) {
+      // warn user
+      // todo: check if create project is in progress
+      const result = dialog.showMessageBox({
+        type: 'warning',
+        message:
+          'There are active tasks. Do you really want to quit?\n\n' +
+          mapActionsToString(activeActions),
+        buttons: ['Abort', 'Yes, proceed (UNSAFE)'],
+      });
+
+      if (result === 0) {
+        ipcRenderer.send('triggerClose', false);
+        return;
+      }
+    }
+
+    ipcRenderer.send('triggerClose', true);
     ipcRenderer.send('killAllRunningProcesses');
   };
 
   render() {
     const { children, isAppLoaded } = this.props;
     const { wasSuccessfullyInitialized } = this.state;
 
+    console.log('app init', this.props.isQueueEmpty);

```

Comment:
I assume this can be removed.

Reply:
Yes, sorry, I missed that one. It's removed.
@superhawk610 What do you think should I open an issue for the exiting during Project creation? I think it's not critical but it could produce partly created projects on disk.
Right now, I can not tackle this as we need the `app-status.reducer`. Once this is available the blocking of exit should be easy to add.